amqp/src/main/scala/akka/stream/alpakka/amqp/AmqpConnectorSettings.scala

```diff
@@ -153,22 +154,52 @@ object TemporaryQueueSourceSettings {
     TemporaryQueueSourceSettings(connectionProvider, exchange)
 }
 
+final class AmqpPublishConfirmConfiguration private (
+    val confirmTimeout: FiniteDuration
+) {
+  override def toString: String =
+    "AmqpPublishConfirmConfiguration(" +
+    s"confirmTimeout=$confirmTimeout" +
+    ")"
+}
+
+object AmqpPublishConfirmConfiguration {
+  def apply(confirmTimeout: FiniteDuration): AmqpPublishConfirmConfiguration =
+    new AmqpPublishConfirmConfiguration(confirmTimeout)
+
+  def unapply(settings: AmqpPublishConfirmConfiguration): Option[FiniteDuration] =
+    Some(settings.confirmTimeout)
+
+  /**
+   * Java API
+   */
+  def create(confirmTimeout: FiniteDuration): AmqpPublishConfirmConfiguration =
+    new AmqpPublishConfirmConfiguration(confirmTimeout)
+}
+
 final class AmqpReplyToSinkSettings private (
     val connectionProvider: AmqpConnectionProvider,
-    val failIfReplyToMissing: Boolean = true
+    val failIfReplyToMissing: Boolean = true,
+    val publishConfirm: Option[AmqpPublishConfirmConfiguration] = None
 ) extends AmqpConnectorSettings {
   override final val declarations = Nil
 
   def withFailIfReplyToMissing(failIfReplyToMissing: Boolean): AmqpReplyToSinkSettings =
     copy(failIfReplyToMissing = failIfReplyToMissing)
 
-  private def copy(connectionProvider: AmqpConnectionProvider = connectionProvider, failIfReplyToMissing: Boolean) =
-    new AmqpReplyToSinkSettings(connectionProvider, failIfReplyToMissing)
+  def withPublishConfirm(confirmTimeout: FiniteDuration): AmqpReplyToSinkSettings =
```

Comment:
Add the corresponding Java API with `java.time.Duration`.

Reply:
With the reasoning of the `Option` above, this should take `AmqpPublishConfirmConfiguration` instead.